src/connector/with_options_sink.yaml

```diff
@@ -365,9 +386,30 @@ PulsarConfig:
   - name: oauth.scope
     field_type: String
     required: false
-  - name: s3_credentials
-    field_type: HashMap<String,String>
-    comments: required keys refer to [`crate::aws_utils::AWS_DEFAULT_CONFIG`]
+  - name: region
+    field_type: String
+    required: false
+  - name: endpoint
+    field_type: String
+    required: false
+    alias: endpoint_url
+  - name: access_key
+    field_type: String
+    required: false
+  - name: secret_key
+    field_type: String
+    required: false
+  - name: session_token
+    field_type: String
+    required: false
+  - name: arn
+    field_type: String
+    required: false
+  - name: external_id
+    field_type: String
+    required: false
```

Comment:
Oops, this field is a new one, not included in `AWS_DEFAULT_CONFIG`

Reply:
BTW, I noticed there are some inconsistencies: 

For kinesis, these are like
```
  - name: aws.credentials.secret_access_key
    field_type: String
    required: false
    alias: kinesis.credentials.secret
```

But for Bigquery and Pulsar, these are simple. Not sure whether/how to improve it.

Reply:
because aws provides more complex auth, I think we miss the `external_id` part for s3 auth.

Reply:
Didn't get it. Do you mean `external_id` is needed or not needed? Do we need to change anything in this PR?

Reply:
`external_id` is an optional field for aws auth. IIRC, we impl it in kinesis and I don't know if it is in the impl for s3 auth.